### PR TITLE
GSGMF-1062: remplace PhantomJS checks with Puppeteer

### PR DESCRIPTION
**Pull request by llienher**

### CODE REVIEW

File: buildtools/check-example.js [hunk 1]
```diff
@@ -1,76 +1,63 @@
...
-if (args.length != 2) {
-  phantom.exit(1);
+const path = require('path');
+const puppeteer = require('puppeteer');
+
+const arg = process.argv[2];
+const screenshotPath = `${arg}.png`;
+const url = 'file:' + path.resolve(__dirname, `../${arg}`);
+if (!url) {
+  throw 'Please provide a URL as the first argument';
```
Comment (sbrunner) [inline buildtools/check-example.js]:
```suggestion
  throw new Error('Please provide an HTML file as the first argument');
```

File: buildtools/check-example.js [hunk 2]
```diff
@@ -1,76 +1,63 @@
...
+    } else {
+      console.log(`Request failed on: ${url}`);
+      error = true;
+    }
+  });
+  page.once('console', (e) => {
+    const type = e.type();
+    if (type !== 'log' && type !== 'debug' && type !== 'info' && type !== 'warning') {
+      console.log('Console error');
+      console.log(e.text());
```
Comment (sbrunner) [inline buildtools/check-example.js]:
```suggestion
      console.log(`${type}: ${e.text()}`);
```

File: buildtools/check-example.js [hunk 3]
```diff
@@ -1,76 +1,63 @@
...
-page.onUrlChanged = function(url) {
-  console.log('URL changed: ' + url);
-};
-page.open(examplePath, function(s) {
-  if (s != 'success') {
-    console.error('PAGE LOAD ERROR');
-    phantom.exit(2);
+(async () => {
+  let error = false;
+  const browser = await puppeteer.launch();
```
Comment (sbrunner) [inline buildtools/check-example.js]: To fix CI:
```suggestion
  const browser = await puppeteer.launch({headless:true,args:['--allow-file-access-from-files']});
```

File: buildtools/check-example.js [hunk 4]
```diff
@@ -1,76 +1,63 @@
...
+  const page = await browser.newPage();
+  await page.setViewport({width: 1920, height: 1080});
+  await page.on('load', () => console.log(`Page loaded: ${page.url()}`));
+  page.once('pageerror', (e) => {
+    console.log('Page error');
+    console.log(e);
+    error = true;
+  });
+  page.once('dialog', (e) => {
+    console.log('Alert');
```
Comment (sbrunner) [inline buildtools/check-example.js]:
```suggestion
    console.log('Unexpected alert message');
```

File: buildtools/check-example.js [hunk 5]
```diff
@@ -1,76 +1,93 @@
...
-    );
-    phantom.exit(exitCode);
-  }, 3000)
-});
+  timeout = setTimeout(async () => {
+    if (requestsURL.size) {
+      console.log('Pending requests:');
+      for (const r of requestsURL) {
+        console.log(r);
+      }
```
Comment (gberaudo) [inline buildtools/check-example.js]: You might want to test: `requestsURL.forEach(console.log)`

File: buildtools/check-example.js [hunk 6]
```diff
@@ -1,76 +1,93 @@
...
+      process.exit(2);
+    } else {
+      await page.screenshot({
+        path: screenshotPath
+      }).then(console.log(`Screenshot saved at: ${screenshotPath}`));
```
Comment (gberaudo) [inline buildtools/check-example.js]:
```suggestion
      }).then(() => console.log(`Screenshot saved at: ${screenshotPath}`));
```

File: buildtools/check-example.js [hunk 7]
```diff
@@ -1,76 +1,93 @@
...
+      // @ts-ignore
+      console.log(`Check finished in ${new Date() - start} seconds`);
+      await browser.close();
```
Comment (gberaudo) [inline buildtools/check-example.js]: You can add a finally block and remove the async / awaits.

File: buildtools/check-example.js [hunk 8]
```diff
@@ -1,76 +1,93 @@
...
+    }
+  }, 2000);
+}
+(async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  await page.setViewport({width: 1920, height: 1080});
+  await page.on('load', () => console.log(`Page loaded: ${page.url()}`));
```
Comment (gberaudo) [inline buildtools/check-example.js]:
```suggestion
  page.on('load', () => console.log(`Page loaded: ${page.url()}`));
```

File: buildtools/check-example.js [hunk 9]
```diff
@@ -1,76 +1,93 @@
...
+    }
+    loaded(page, browser);
+  });
+  page.on('console', (message) => {
+    const type = message.type();
+    const location = message.location();
+    if (type !== 'log' && type !== 'debug' && type !== 'info' && type !== 'warning' &&
+      location.url != 'http://localhost:3000/.build/examples-hosted/dist/vendor.js?dev'
+    ) {
+      console.log('Console error');
```
Comment (gberaudo) [inline buildtools/check-example.js]:
```suggestion
      console.log(`Console ${type}`);
```

File: buildtools/check-example.js [hunk 10]
```diff
@@ -1,76 +1,93 @@
...
+      console.log(`On: ${location.url} ${location.lineNumber}:${location.columnNumber}.`);
+      console.log(`${type}: ${message.text()}`);
```
Comment (gberaudo) [inline buildtools/check-example.js]:
```suggestion
      console.log(message.text());
```

**Pull request merged**